windowing.py

```diff
@@ -25,103 +28,175 @@ def create_windows_from_dictmatrix(dict_matrix, rows, window_size, stride):
+
+    if (return_lift_windows):
+
+        return get_lift_windows_from_indices(indices, windowed_data)
+
     else:
-        indices = np.squeeze(indices)
-        count = indices[0]-1
-        for i in range(len(indices)):
-            count += 1
-            if count != indices[i]:
-                end = i
-                break
-        indices = indices[:i]
-    lift_windows = np.take(windows, indices, axis=0)
-    return lift_windows
+        if (indices.size == 0):
+            return None
+        index_of_start = indices[0]
+        start_time = windows[index_of_start][0][3][0]
+        return start_time
+
+
+def get_lift_windows_from_indices(indices, windowed_data):
+	"""
+    Description: Finds the first set of continuous windows from the indices array
 
+    Args:
+        windowed_data -- np array
+        indices -- int (threshold for variance, if using a divisor set to 0 instead)
 
-def refine_start_time(windowed_data, threshold):
-    # Description: finds the more refined start time by calcualting variacne
-    #				on the smaller windows and returning the start time of the first
-    #				window that meets the thresholdusing a smaller threshold.
-    #
-    # Input: windowed_data = np array
-    # Input: threshold = int (variance threshold for start of lift
-    #
-    #
-    # Return: refined start time = int (in nanos)
+    Return:
+        lift_windows -- np_array (array of indices that correspond to windows in
+                       windowed_data that meet the variance threshold level)
+    """
+	windows = windowed_data[0]
+	if (indices.size <= 1):
+		indices = indices[0]
+	else:
+		indices = np.squeeze(indices)
+		indices_diff = np.diff(indices)
+		end_time = np.argwhere(indices_diff != 1)
+		if (end_time.size != 0):
+			indices = indices[:end_time[0][0]+1]
+	lift_windows = np.take(windows, indices, axis=0)
+	return lift_windows
+
+def find_indices_that_meet_threshold(windowed_data, threshold, divisor):
+    """
+    Description: Finds indices that meet the desired threshold level
+
+    Args:
+        windowed_data -- np array
+        threshold -- int (threshold for variance, if using a divisor set to 0 instead)
+        divisor -- int (divisor to set threshold, if using threshold set to 0 instead)
+
+
+    Return:
+        indices -- np_array (array of indices that correspond to windows in
+                   windowed_data that meet the variance threshold level)
+    """
     windows = windowed_data[0]
-    windows_no_nanos = np.delete(windowed_data[0], 3, 1)
+    windows_no_nanos = windows[:, :3]
     variances = np.var(windows_no_nanos, axis=2, ddof=1)
     var_sum = np.sum(variances, axis=1)
-    indexes = np.argwhere(var_sum > threshold)
-    if (len(indexes) < 1):
-        return None
-    index_of_start = indexes[0]
-    return windows[index_of_start][0][3][0]
+    if divisor == 0:
+        indices = np.argwhere(var_sum > threshold)
+    else:
+        indices = np.argwhere(var_sum > max(var_sum)/divisor)
```

Comment:
similar to above. Try to avoid different calls to the same function if just the value of the parameter is changed.